src/simulation/simulation.jl

```diff
@@ -656,89 +700,67 @@ function _execute!(
-        container = get_optimization_container(model)
-        num_rows = num_executions * get_steps(sim)
-
-        # TODO: configuration of keep_in_cache and priority are not correct
+"""
+    execute!(sim::Simulation; kwargs...)
 
-        for (key, array) in get_duals(container)
-            reqs.duals[model] = _calc_dimensions(array, encode_key(key), num_rows, horizon)
-            add_rule!(rules, model_name, key, false, CachePriority.LOW)
-        end
+Solves the simulation model for sequential Simulations.
 
-        for (key, param_container) in get_parameters(container)
-            array = get_parameter_array(param_container)
-            reqs.parameters[key] =
-                _calc_dimensions(array, encode_key(key), num_rows, horizon)
-            add_rule!(rules, model_name, key, false, CachePriority.LOW)
-        end
+# Arguments
+- `sim::Simulation=sim`: simulation object created by Simulation()
 
-        for (key, array) in get_variables(container)
-            reqs.variables[key] =
-                _calc_dimensions(array, encode_key(key), num_rows, horizon)
-            add_rule!(rules, model_name, key, false, CachePriority.HIGH)
-        end
+The optional keyword argument `exports` controls exporting of results to CSV files as
+the simulation runs. Refer to [`export_results`](@ref) for a description of this argument.
 
-        for (key, array) in get_aux_variables(container)
-            reqs.aux_variables[key] =
-                _calc_dimensions(array, encode_key(key), num_rows, horizon)
-            add_rule!(rules, model_name, key, false, CachePriority.HIGH)
-        end
+# Example
+```julia
+sim = Simulation("Test", 7, problems, "/Users/folder")
+execute!(sim::Simulation; kwargs...)
+```
+"""
+function execute!(sim::Simulation; kwargs...)
+    file_mode = "a"
+    logger = configure_logging(sim.internal, file_mode)
+    register_recorders!(sim.internal, file_mode)
 
-        # TODO: Do for expressions
-        #for (key, array) in get_expressions(model)
-        #    reqs.aux_variables[key] =
-        #        _calc_dimensions(array, encode_key(key), num_rows, horizon)
-        #    add_rule!(
-        #        rules,
-        #        model_name,
-        #        key,
-        #        false,
-        #        CachePriority.HIGH,
-        #    )
-        #end
+    # Undocumented option for test & dev only.
+    in_memory = get(kwargs, :in_memory, false)
+    store_type = in_memory ? InMemorySimulationStore : HdfSimulationStore
 
-        model_req[model_name] = reqs
+    if (get_simulation_build_status(sim) != BuildStatus.BUILT) ||
+       (get_simulation_status(sim) != RunStatus.READY)
+        error("Simulation status is invalid, you need to rebuild the simulation")
+    end
+    try
+        open_store(store_type, get_store_dir(sim), "w") do store
+            set_simulation_store!(sim, store)
+            # TODO: return file name for hash calculation instead of hard code
```

Comment:
Is this comment still relevant?

Reply:
I think this has been there for a while, do you recall what this is about? it seems related to the set of the store file